### PR TITLE
:arrow_up: refactor code to work with despina

### LINKED ISSUE #13: Upgrade to Despina
We did a big refactor in go-perun and moved connection handling out of the `client.Client` into a `wire/net.Bus`. The tcp/unix dialer and listener implementations were moved into `wire/net/simple`. This repo should be updated:
- [ ] Before creating the client, create a `net.Bus` with a tcp dialer and listener.
- [ ] Start `Listen`ing with the `net.Bus`, not with the client.
- [ ] Pass this bus to the client's constructor instead of a tcp dialer and listener, which were given to the bus before.
- [ ] `Client.Reconnect` was renamed to `Restore` and the meaning of this method changed. Use it when persistence is active.

Maybe I forgot something, but this should become apparent while testing.

### PR DESCRIPTION
**Pull request by matthiasgeihs**

### CODE REVIEW

File: cmd/demo/node.go [hunk 1]
```diff
@@ -115,7 +113,11 @@ func (n *node) peer(addr perunpeer.Address) *peer {
 }
 
 func (n *node) setupChannel(ch *client.Channel) {
-	perunID := ch.Peers()[0]
+	if len(ch.Peers()) > 2 {
```
Comment (ggwpez) [inline cmd/demo/node.go]: I think this check does not make sense, since *go-perun* can only do two-party channels in *Despina*.  
The version of *go-perun* is fixed anyway, so there is no way this changes unless we update it. If you still want to have the check, you also need to do it in the `HandleProposal` function.

File: cmd/demo/node.go [hunk 2]
```diff
@@ -25,23 +25,25 @@ import (
 	"perun.network/go-perun/channel"
 	"perun.network/go-perun/client"
 	"perun.network/go-perun/log"
-	perunpeer "perun.network/go-perun/peer"
-	"perun.network/go-perun/peer/net"
 	"perun.network/go-perun/wallet"
+	"perun.network/go-perun/wire"
+	wirenet "perun.network/go-perun/wire/net"
+	wirenetsimple "perun.network/go-perun/wire/net/simple"
```
Comment (ggwpez) [inline cmd/demo/node.go]:
```suggestion
	simple "perun.network/go-perun/wire/net/simple"
```
We only use names like `ethchannel` sometimes because there are multiple `channel`. For the `simple` there is only one so it should be fine just naming it `simple`.

**Review by ggwpez — CHANGES_REQUESTED** [review]
Looks good

File: cmd/demo/node.go (hunk 1 above)
Comment (matthiasgeihs) [reply cmd/demo/node.go]: i vote for including both checks, when initiating a channel opening and when receiving a proposal. the reason is that a developer who uses our code should be aware that the code only works for the two-party case. this is why i added also comments on the respective lines of code. i also changed the action from warning to fail, because code execution should not continue running if for some reason we have other than 2 participants in the channel.

File: cmd/demo/node.go (hunk 2 above)
Comment (matthiasgeihs) [reply cmd/demo/node.go]: ok, the reasoning was that "simple" is a bit of a too simple name to be understood what this refers to when out of context. but i guess the context is given and, as you suggest, only using package aliases when there is a package name overload seems to be the standard way to do it.

File: cmd/demo/node.go (hunk 1 above)
Comment (ggwpez) [reply cmd/demo/node.go]: Well, the code is not meant to be used by other developers since we do not have docs or tests (and it is just a demo). It also does not work for sub-channels, so another test to check `Locked` for zero-sum would be needed or a comment.

File: cmd/demo/node.go [hunk 3]
```diff
@@ -115,7 +113,11 @@ func (n *node) peer(addr perunpeer.Address) *peer {
 }
 
 func (n *node) setupChannel(ch *client.Channel) {
-	perunID := ch.Peers()[0]
+	if len(ch.Peers()) != 2 {
+		log.Fatal("Channels with more than 2 participants are currently not supported")
```
Comment (ggwpez) [inline cmd/demo/node.go]:
```suggestion
		log.Fatal("Only channels with two participants are currently supported")
```

File: cmd/demo/node.go [hunk 4]
```diff
@@ -187,6 +189,10 @@ func (n *node) channel(id channel.ID) *paymentChannel {
 }
 
 func (n *node) HandleProposal(req *client.ChannelProposal, res *client.ProposalResponder) {
+	if len(req.PeerAddrs) != 2 {
+		log.Fatal("Channels with more than 2 participants are currently not supported")
```
Comment (ggwpez) [inline cmd/demo/node.go]:
```suggestion
		log.Fatal("Only channels with two participants are currently supported")
```

Comment (matthiasgeihs) [reply cmd/demo/node.go]: done

File: cmd/demo/node.go (hunk 3 above)
Comment (matthiasgeihs) [reply cmd/demo/node.go]: done

File: cmd/demo/node_setup.go [hunk 5]
```diff
@@ -40,8 +39,8 @@ var (
 // configuration from viper.
 func Setup() {
 	SetConfig()
-	rng := rand.New(rand.NewSource(0x280a0f350eec))
-	appDef := wtest.NewRandomAddress(rng)
+
+	appDef := ewallet.AsWalletAddr(common.Address{}) // dummy app def
```
Comment (ggwpez) [inline cmd/demo/node_setup.go]:
```suggestion
	appDef := &ewallet.Address{} // dummy app def
``` 
or just remove the line and do `payment.SetAppDef(&ewallet.Address{})`.

**Review by ggwpez — CHANGES_REQUESTED** [review]
Please just squash the commits, looks good otherwise.

File: cmd/demo/node_setup.go (hunk 5 above)
Comment (matthiasgeihs) [reply cmd/demo/node_setup.go]: done

**Review by ggwpez — DISMISSED** [review]

**Comment by ggwpez** [comment]
I will quickly rebase the branch and then merge.

**Review by ggwpez — APPROVED** [review]
